### PR TITLE
 Profile skeleton (EXPOSUREAPP-12664)

### PR DESCRIPTION
**Pull request by chiljamgossow**
profile repository interface

### CODE REVIEW

**Review by SamuraiKek — DISMISSED** [review]

**Review by SamuraiKek — DISMISSED** [review]

**Review by CV113 — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/model/Profile.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package de.rki.coronawarnapp.profile.model
+
+import org.joda.time.LocalDate
+
+data class Profile(
+    val id: String? = null,
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/model/Profile.kt:6]: is it really `optional` if we have a profile all fields could be optional ,but we still have a profile

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/storage/ProfileRepository.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import de.rki.coronawarnapp.profile.model.Profile
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import org.joda.time.LocalDate
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class ProfileRepository @Inject constructor() {
+    val profileFlow: Flow<Set<Profile>> = flowOf(setOf(dummy1, dummy2))
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/storage/ProfileRepository.kt:12]: `profilesFlow` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragmentViewModel.kt [hunk 3]
```diff
@@ -6,31 +6,40 @@ import androidx.lifecycle.asLiveData
...
 class RATProfileCreateFragmentViewModel @AssistedInject constructor(
-    private val ratProfileSettings: RATProfileSettingsDataStore,
+    private val profileRepository: ProfileRepository,
     @Assisted private val format: DateTimeFormatter = DateTimeFormat.mediumDate()
 ) : CWAViewModel() {
 
-    private val profileData = MutableLiveData(RATProfileData())
-    val profile: LiveData<RATProfileData> = profileData
+    // TO DO get id as nav arg
+    private val id: String? = "1"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragmentViewModel.kt:26]: it should not be optional specially  when needed in navigation

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/model/Profile.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/model/Profile.kt:6]: This is for new profiles, the id will be assigned on insert

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragmentViewModel.kt (hunk 3 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragmentViewModel.kt:26]: this is for creating new profiles, then the id is null

**Review by mtwalli — APPROVED** [review]

**Review by SamuraiKek — APPROVED** [review]

**Pull request merged**